braidpool-primitives/src/utils/test_utils.rs

```diff
@@ -0,0 +1,152 @@
+#![allow(unused)]
+use crate::beads::{Bead, TransactionWithMerklePath};
+use crate::utils::BeadHash;
+use crate::utils::bitcoin::MerklePathProof;
+use ::bitcoin::absolute::LockTime;
+use ::bitcoin::absolute::Time;
+use ::bitcoin::hex::FromHex;
+use ::bitcoin::{Amount, Txid};
+use ::bitcoin::{
+    BlockHash, BlockHeader, BlockTime, BlockVersion, CompactTarget, OutPoint, ScriptBuf, Sequence,
+    Transaction, TransactionVersion, TxIn, TxMerkleNode, TxOut, Witness,
+};
+use std::collections::HashSet;
+pub fn create_block_header(
+    version: i32,
+    prev_blockhash_bytes: [u8; 32],
+    merkle_root_bytes: [u8; 32],
+    time: u32,
+    bits: u32,
+    nonce: u32,
+) -> BlockHeader {
+    BlockHeader {
+        version: BlockVersion::from_consensus(version),
+        prev_blockhash: BlockHash::from_byte_array(prev_blockhash_bytes),
+        merkle_root: TxMerkleNode::from_byte_array(merkle_root_bytes),
+        time: BlockTime::from_u32(time),
+        bits: CompactTarget::from_consensus(bits),
+        nonce,
+    }
+}
+
+pub fn create_dummy_transaction(
+    bytes: [u8; 32],
+    vout: u32,
+    sequence: u32,
+    witness_items: Vec<Vec<u8>>,
+    value: u64,
+) -> Transaction {
+    Transaction {
+        version: TransactionVersion::TWO,
+        lock_time: LockTime::ZERO,
+        input: vec![TxIn {
+            previous_output: OutPoint {
+                txid: Txid::from_byte_array(bytes),
+                vout,
+            },
+            script_sig: ScriptBuf::new(),
+            sequence: Sequence(sequence),
+            witness: Witness::from_slice(&witness_items),
+        }],
+        output: vec![TxOut {
+            value: Amount::from_sat(value).unwrap(),
+            script_pubkey: ScriptBuf::new(),
+        }],
+    }
+}
+pub fn create_dummy_merkle_path_proof(bytes: [u8; 32]) -> MerklePathProof {
+    MerklePathProof {
+        transaction_hash: Txid::from_byte_array(bytes),
+        merkle_path: vec![],
+    }
+}
+pub fn create_parents_set(parents_vec: Vec<(BeadHash, Time)>) -> HashSet<(BeadHash, Time)> {
+    parents_vec.into_iter().collect()
+}
+pub fn create_test_bead(
+    block_version: i32,
+    prev_blockhash_bytes: [u8; 32],
+    merkle_root_bytes: [u8; 32],
+    timestamp: u32,
+    difficulty_bits: u32,
+    nonce: u32,
+    bead_hash_bytes: [u8; 32],
+    coinbase_tx_bytes: [u8; 32],
```

Comment:
```suggestion
    nonce: u32,
    coinbase_tx_bytes: [u8; 32],
```
If I'm not wrong, `bead_hash_bytes` is not used at all and the `bead_hash` is in the `bead_bytes`.